arangod/Utils/Cursor.h

```diff
@@ -85,6 +92,30 @@ class Cursor {
 
   void setDeleted() { _isDeleted = true; }
 
+  void setLastQueryBatchObject(
+      std::shared_ptr<velocypack::Buffer<uint8_t>> buffer) {
+    _currentBatchResult.second = buffer;
```

Comment:
The shared_ptr is passed by value, so it can be moved here. Otherwise there will be 2 copies of it.
```suggestion
    _currentBatchResult.second = std::move(buffer);
```